rmw_zenoh_cpp/src/detail/rmw_subscription_data.cpp

```diff
@@ -505,10 +505,10 @@ void SubscriptionData::add_new_message(
       msg->attachment.sequence_number() -
       last_known_pub_it->second);
     if (seq_increment > 1) {
-      const size_t num_msg_lost = seq_increment - 1;
+      const int64_t num_msg_lost = seq_increment - 1;
       events_mgr_->update_event_status(
         ZENOH_EVENT_MESSAGE_LOST,
-        num_msg_lost);
+        static_cast<int32_t>(num_msg_lost));
```

Comment:
The truncation here from the static_cast can lead to UB if `num_msg_lost` is greater than `INT32_MAX`. It's unlikely that we'll ever have those many lost messages but it's probably better if we can avoid such patterns since we're fixing the implementation here.

How about

1. Switch to `int64_t` as the datatype for [current_count_change](https://github.com/ros2/rmw_zenoh/blob/bb2deedea45f45621ed2e6cc99874a8e011f4cdc/rmw_zenoh_cpp/src/detail/event.hpp#L120)
2. Pass `num_msg_lost` as is here

```suggestion
        num_msg_lost);
```

3. Update the implementation of [update_event_status](https://github.com/ros2/rmw_zenoh/blob/bb2deedea45f45621ed2e6cc99874a8e011f4cdc/rmw_zenoh_cpp/src/detail/event.cpp#L172) to clamp the value of `current_count_change` to [INT32_MIN, INT32_MAX]